src/lib/components/tablev2/MultiSelectableContent.tsx

```diff
@@ -0,0 +1,281 @@
+import { useEffect, memo, useCallback, useState, CSSProperties } from 'react';
+import { Row } from 'react-table';
+import { areEqual } from 'react-window';
+
+import { ConstrainedText } from '../constrainedtext/Constrainedtext.component';
+import { Tooltip } from '../tooltip/Tooltip.component';
+import { useTableContext } from './Tablev2.component';
+import {
+  HeadRow,
+  NoResult,
+  SortCaretWrapper,
+  SortIncentive,
+  TableBody,
+  TableHeader,
+  TableRowMultiSelectable,
+  TooltipContent,
+  UnknownIcon,
+} from './Tablestyle';
+import {
+  TableHeightKeyType,
+  TableLocalType,
+  TableVariantType,
+  VirtualizedRows,
+} from './TableUtils';
+
+const translations = {
+  en: {
+    noResult: 'No results found',
+  },
+  fr: {
+    noResult: `Aucun résultat`,
+  },
+};
+
+type RenderRowType = {
+  index: number;
+  style: CSSProperties;
+};
+
+type MultiSelectableContentProps = {
+  onMultiSelectionChanged: (rows: Row<object>[]) => void;
+  rowHeight?: TableHeightKeyType;
+  separationLineVariant?: TableVariantType;
+  backgroundVariant?: TableVariantType;
+  locale?: TableLocalType;
+  customItemKey?: (index: Number, data: any) => string;
+  children?: (rows: JSX.Element) => JSX.Element;
+};
```

Comment:
I would try to make this type generic so that the row object can actually be typed
```suggestion
type MultiSelectableContentProps<ENTRY extends Record<string, any>> = {
  onMultiSelectionChanged: (rows: Row<ENTRY>[]) => void;
  rowHeight?: TableHeightKeyType;
  separationLineVariant?: TableVariantType;
  backgroundVariant?: TableVariantType;
  locale?: TableLocalType;
  customItemKey?: (index: Number, data: ENTRY) => string;
  children?: (rows: JSX.Element) => JSX.Element;
};
```

Reply:
I think locale is a required props when the children props is not passed so the type could be reviewed in the following way:

```suggestion
type MultiSelectableContentProps<ENTRY extends Record<string, any>> = {
  onMultiSelectionChanged: (rows: Row<ENTRY>[]) => void;
  rowHeight?: TableHeightKeyType;
  separationLineVariant?: TableVariantType;
  backgroundVariant?: TableVariantType;
  customItemKey?: (index: Number, data: ENTRY) => string;
} & ({
  locale: TableLocalType;
} | {
  children: (rows: JSX.Element) => JSX.Element;
  });
```

Reply:
I've encounter some issues tackling this, so I put a comment and created a ticket for that. I will resume this work in a future PR ASAP